velox/common/file/FileInputStream.cpp

```diff
@@ -185,6 +185,26 @@ void FileInputStream::readBytes(uint8_t* bytes, int32_t size) {
   }
 }
 
+std::unique_ptr<folly::IOBuf> FileInputStream::readBytes(int32_t size) {
+  VELOX_CHECK_GE(size, 0, "Attempting to read negative number of bytes");
+  if (size == 0) {
+    return nullptr;
+  }
+
+  VELOX_CHECK_LE(
+      size, remainingSize(), "Read past the end of input file {}", fileSize_);
+  if (current_->availableBytes() >= size) {
+    auto iobuf =
+        folly::IOBuf::wrapBuffer(current_->buffer + current_->position, size);
+    current_->position += size;
+    return iobuf;
+  }
+  auto iobuf = folly::IOBuf::create(size);
```

Comment:
This allocation should be avoided.  Can we make a chain that `takeOwnership()` of the buffers instead?  This would also avoid new calls to this function invalidate the old buffers.

Reply:
I think it is complicated, and it breaks the FileInputStream should only reserve memory as QueryConfig::readBufferSize. Alternatively, we could add a new argument `MemoryPool* pool = nullptr`, allocated the Buffer from pool, and wrap the buffer as IOBuf. Which way do you think is better? This maybe in a follow up PR.